src/cdi_CPEloss/Tabular_CP_Eloss.cc

```diff
@@ -0,0 +1,346 @@
+ *                 separating the stopping power approximation from analog 
+ *                 scattering
+ */
+Tabular_CP_Eloss::Tabular_CP_Eloss(
+    const std::string &filename_in, rtt_cdi::CParticle target_in,
+    rtt_cdi::CParticle projectile_in,
+    rtt_cdi::CPModelAngleCutoff model_angle_cutoff_in)
+    : rtt_cdi::CPEloss(target_in, projectile_in,
+                       rtt_cdi::CPModelType::TABULAR_ETYPE,
+                       model_angle_cutoff_in),
+      //filename(filename_in), us(rtt_units::UnitSystemType().X4()), pc_cgs(us) {
+      filename(filename_in) {
+  using std::stod;
+  using std::stoi;
+
+  file.open(filename);
+  Insist(file.is_open(), "Error opening DEDX file \"" + filename + "\"");
+
+  constexpr uint32_t max_entries =
+      6; // This is a statement about the file format, maximum of six entries per row.
+
+  std::vector<std::string> line_entries = read_line(); // ZAID
+  int32_t projectile_zaid_file = stoi(line_entries[0]);
+  Require(projectile.get_zaid() == projectile_zaid_file);
+
+  skip_lines(1); // Z, A, mass
+
+  line_entries = read_line(); // Number of bins for energy, density, temperature
+  Check(line_entries.size() >= 3);
+  n_energy = stoi(line_entries[0]);
+  n_density = stoi(line_entries[1]);
+  n_temperature = stoi(line_entries[2]);
+
+  line_entries =
+      read_line(); // Bin spacing for energy, density, temperature (log)
+  Check(line_entries.size() >= 3);
+  d_log_energy = stod(line_entries[0]);
+  d_log_density = stod(line_entries[1]);
+  d_log_temperature = stod(line_entries[2]);
+  Require(d_log_energy > 0);
+  Require(d_log_density > 0);
+  Require(d_log_temperature > 0);
+  d_log_energy = 1. / d_log_energy;
+  d_log_density = 1. / d_log_density;
+  d_log_temperature = 1. / d_log_temperature;
+
+  // Get first energy support point
+  uint32_t nlines = (n_energy + max_entries - 1) / max_entries;
+  line_entries = read_line();
+  min_log_energy = stod(line_entries[0]);
+  skip_lines(nlines - 1);
+  energies.resize(n_energy);
+  for (uint32_t n = 0; n < n_energy; n++) {
+    energies[n] = exp(min_log_energy + n * d_log_energy);
+  }
+
+  // Get first density support point
+  nlines = (n_density + max_entries - 1) / max_entries;
+  line_entries = read_line();
+  min_log_density = stod(line_entries[0]);
+  skip_lines(nlines - 1);
+  densities.resize(n_density);
+  for (uint32_t n = 0; n < n_density; n++) {
+    densities[n] = exp(min_log_density + n * d_log_density);
+  }
+
+  // Get first temperature support point
+  nlines = (n_temperature + max_entries - 1) / max_entries;
+  line_entries = read_line();
+  min_log_temperature = stod(line_entries[0]);
+  skip_lines(nlines - 1);
+  temperatures.resize(n_temperature);
+  for (uint32_t n = 0; n < n_temperature; n++) {
+    temperatures[n] = exp(min_log_temperature + n * d_log_temperature);
+  }
+
+  std::vector<double> stopping_data_1d(n_energy * n_density * n_temperature);
```

Comment:
This is memory error.  `mdspan` is only a _view_ to this data.   Once the constructor goes out of scope this storage is invalid.  You need to move `stopping_data_1d` to the class definition.  This is why the Win32 test was failing.